src/vs/workbench/contrib/debug/browser/disassemblyView.ts

```diff
@@ -0,0 +1,209 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { getPixelRatio, getZoomLevel } from 'vs/base/browser/browser';
+import { Dimension, append, $ } from 'vs/base/browser/dom';
+import { ITableRenderer, ITableVirtualDelegate } from 'vs/base/browser/ui/table/table';
+import { BareFontInfo } from 'vs/editor/common/config/fontInfo';
+import { localize } from 'vs/nls';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { WorkbenchTable } from 'vs/platform/list/browser/listService';
+import { IStorageService } from 'vs/platform/storage/common/storage';
+import { ITelemetryService } from 'vs/platform/telemetry/common/telemetry';
+import { editorBackground } from 'vs/platform/theme/common/colorRegistry';
+import { IThemeService } from 'vs/platform/theme/common/themeService';
+import { EditorPane } from 'vs/workbench/browser/parts/editor/editorPane';
+import { EditorInput } from 'vs/workbench/common/editor/editorInput';
+import { DISASSEMBLY_VIEW_ID } from 'vs/workbench/contrib/debug/common/debug';
+
+interface IDisassembledInstructionEntry {
+	allowBreakpoint: boolean;
+	isBreakpointSet: boolean;
+	instruction: DebugProtocol.DisassembledInstruction;
+}
+
+export class DisassemblyView extends EditorPane {
+
+	private static readonly NUM_INSTRUCTIONS_TO_LOAD = 50;
```

Comment:
50 is a very low number here. The scroll bar will considerably jump as soon as the user scrolls a bit.
I suggest to use a much larger number like a 1000 and make sure to have a lazily model that is evaluated only once the user scrolls all the way up to it (the element gets rendered)

Reply:
In fact we don't want this number to be too large. The code may run on remote devices than doesn't have a very large communication bandwidth. I agree we can re-evaluate this value later though.

Reply:
The table scrolling will be very weird and the UI wont be smooth. It wont be a nice user experience.
If remote is the only concern I suggest to maybe even change this number depending if we are in a remote scenario.

Reply:
I think the UI looks fine though. Either way it's only a one line change, and later we can setup a meeting to go through the end to end experience, in which we can fix all of these minor issues.